plc4j/spi/src/main/java/org/apache/plc4x/java/spi/messages/DefaultPlcWriteRequest.java

```diff
@@ -194,6 +196,9 @@ public PlcWriteRequest build() {
                 PlcTag tag = tagValues.getLeft().get();
                 Object[] value = tagValues.getRight();
                 PlcValue plcValue = valueHandler.newPlcValue(tag, value);
+                if (tag.getPlcValueType() == PlcValueType.NULL && value!=null) {
+                    tag.setPlcValueType(plcValue.getPlcValueType());
+                }
```

Comment:
I still don't think we should have this in here ... if the tag says it's one type, but the PlcValue says it's something else, we shouldn't update this (I guess this is the reason you needed to add the "setPlcValue"). We should fire an exception, because in this case the user is doing something incorrectly.

Reply:
![image](https://user-images.githubusercontent.com/1769060/220867841-3be0d2e4-d740-44cd-aaaf-fc2ae2b58d2a.png)

I want to change plc4x tagAddress can praser other driver. eg:  %DB2:94:STRING(10), %DB2:78:LREAL[2]

Reply:
It can remove tag formate that depends on other driver's label formats.
I can remove this change. Next PR. I want to fix plc4x driver and server

Reply:
Please remove this from this PR for now as we have to deal with this "feature" separately.